src/test/ui/issues/issue-27282-move-match-input-into-guard.rs

```diff
@@ -24,7 +24,6 @@ fn main() {
     match b {
         &mut false => {},
         _ if { (|| { let bar = b; *bar = false; })();
-                     //~^ ERROR cannot move out of `b` because it is borrowed [E0505]
```

Comment:
Hmm. It *might* be worthwhile to make variants of this test case that instead of using `_` in the pattern, use `ref _unused` and `ref mut _unused`, just to make sure we cover those cases in our test suite.

Reply:
(or do we already have cases for that elsewhere ....?)